src/pages/ProductDetail/BarChart.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import Chart from 'chart.js/auto';
+
+const StyledCanvas = styled.canvas`
+  width: ${props => props.width};
+  height: ${props => props.height};
+  margin: 25px auto 0 auto;
+`;
+
+const BarChart = ({ width, height }) => {
+  const chartRef = useRef(null);
+
+  useEffect(() => {
+    const myChartRef = chartRef.current.getContext('2d');
+
+    new Chart(myChartRef, {
+      type: 'bar',
+      data: {
+        // labels와 datasets 프로퍼티 설정
+        labels: ['10대', '20대', '30대', '40대'],
+        datasets: [
+          {
+            label: '연령별',
+            data: [8.2, 43.4, 28.6, 13.6],
+            backgroundColor: 'rgba(229, 190, 236, 0.6)',
+            borderColor: 'rgba(229, 190, 236, 1)',
+            borderWidth: 1,
+          },
+        ],
+      },
+      options: {
+        plugins: {
+          ChartDataLabels: true,
+        },
+        tooltip: {
+          enabled: false,
+        },
+        responsive: false,
+        maintainAspectRatio: false,
+        maxBarThickness: 40,
+        scales: {
+          x: {
+            stacked: true,
+            display: true,
+          },
+          y: {
+            stacked: true,
+            display: false,
+            beginAtZero: true,
+          },
+        },
+        categorySpacing: 0,
+      },
+    });
```

Comment:
```suggestion
        const lablesAndDataSets = {
      type: 'bar',
      data: {

        labels: ['10대', '20대', '30대', '40대'],
        datasets: [
          {
            label: '연령별',
            data: [8.2, 43.4, 28.6, 13.6],
            backgroundColor: 'rgba(229, 190, 236, 0.6)',
            borderColor: 'rgba(229, 190, 236, 1)',
            borderWidth: 1,
          },
        ],
      },
      options: {
        plugins: {
          ChartDataLabels: true,
        },
        tooltip: {
          enabled: false,
        },
        responsive: false,
        maintainAspectRatio: false,
        maxBarThickness: 40,
        scales: {
          x: {
            stacked: true,
            display: true,
          },
          y: {
            stacked: true,
            display: false,
            beginAtZero: true,
          },
        },
        categorySpacing: 0,
      },
    }
    new Chart(myChartRef, lablesAndDataSets);
```
주석으로 표현할 수 있는건 변수에다가 담아줄 수 있습니다.